reactor-core/src/jcstress/java/reactor/core/publisher/SinkManyBestEffortStressTest.java

```diff
@@ -0,0 +1,217 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.core.publisher;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Arbiter;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.I_Result;
+import org.openjdk.jcstress.infra.results.LI_Result;
+import org.reactivestreams.Subscription;
+
+import static org.openjdk.jcstress.annotations.Expect.*;
+
+public abstract class SinkManyBestEffortStressTest {
+
+	@JCStressTest
+	@Outcome(id = {"6"}, expect = ACCEPTABLE, desc = "Six parallel subscribes")
+	@State
+	public static class ParallelSubscribersStressTest {
+
+		final SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		@Actor
+		public void one() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Actor
+		public void two() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Actor
+		public void three() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Actor
+		public void four() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Actor
+		public void five() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Actor
+		public void six() {
+			sink.subscribe(new StressSubscriber<>());
+		}
+
+		@Arbiter
+		public void arbiter(I_Result r) {
+			r.r1 = sink.currentSubscriberCount();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {"0"}, expect = ACCEPTABLE, desc = "No subscriber counted")
+	@State
+	public static class CancelOnSubscribeOneSubscribersStressTest {
+
+		final SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		@Actor
+		public void one() {
+			sink.subscribe(new BaseSubscriber<Integer>() {
+				@Override
+				protected void hookOnSubscribe(Subscription subscription) {
+					subscription.cancel();
+				}
+			});
+		}
+
+		@Arbiter
+		public void arbiter(I_Result r) {
+			r.r1 = sink.currentSubscriberCount();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {"1"}, expect = ACCEPTABLE, desc = "One subscriber counted")
+	@State
+	public static class CancelOnSubscribeTwoSubscribersStressTest {
+
+		final SinkManyBestEffort<Integer> sink = SinkManyBestEffort.createBestEffort();
+
+		@Actor
+		public void one() {
+			sink.subscribe(new StressSubscriber<>(0));
+		}
+
+		@Actor
+		public void two() {
+			sink.subscribe(new BaseSubscriber<Integer>() {
+				@Override
+				protected void hookOnSubscribe(Subscription subscription) {
+					subscription.cancel();
+				}
+			});
+		}
+
+		@Arbiter
+		public void arbiter(I_Result r) {
+			r.r1 = sink.currentSubscriberCount();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {"0"}, expect = ACCEPTABLE, desc = "No subscriber counted")
+	@State
+	public static class CancelledBeforeSubscribeOneSubscriberStressTest {
```

Comment:
the whole point of JCstress is to run actors in parallel, so it won't always be "before", but before, at the same time and after the subscription

Reply:
will rename